zone/zonedb.cpp

```diff
@@ -814,22 +815,27 @@ bool ZoneDatabase::LoadCharacterLeadershipAA(uint32 character_id, PlayerProfile_
 }
 
 bool ZoneDatabase::LoadCharacterDisciplines(uint32 character_id, PlayerProfile_Struct* pp){
-	std::string query = StringFormat(
-		"SELECT				  "
-		"disc_id			  "
-		"FROM				  "
-		"`character_disciplines`"
-		"WHERE `id` = %u ORDER BY `slot_id`", character_id);
-	auto results = database.QueryDatabase(query);
-	int i = 0;
 
+	auto character_disciplines = CharacterDisciplinesRepository::GetWhere(
+		database, fmt::format(
+			"`id` = {} ORDER BY `slot_id`",
+			character_id
+		)
+	);
+
+	if (character_disciplines.empty()) {
+		return false;
+	}
+
+	int i = 0;
 	/* Initialize Disciplines */
 	memset(pp->disciplines.values, 0, (sizeof(pp->disciplines.values[0]) * MAX_PP_DISCIPLINES));
-	for (auto& row = results.begin(); row != results.end(); ++row) {
-		if (i < MAX_PP_DISCIPLINES)
-			pp->disciplines.values[i] = atoi(row[0]);
-        ++i;
-    }
+	for (auto& row : character_disciplines) {
+		if (i < MAX_PP_DISCIPLINES && IsValidSpell(row.disc_id)) {
```

Comment:
Shouldn't `row.slot_id` be the thing we're checking here and get rid of `i` altogether ?

Reply:
yes, that should be fine. let me do that.